### PR TITLE
INTC-183 Add support for nexus container analysis

### PR DESCRIPTION
**Pull request by bigspotteddog**
Add nexus container analysis support to the nexus-platform-plugin.

Jira: https://issues.sonatype.org/browse/INTC-183
Jenkins: https://jenkins.ci.sonatype.dev/job/integrations/job/jenkins/job/feature-snapshots/job/INTC-183-container-for-jenkins/

### CODE REVIEW

File: src/main/java/org/sonatype/nexus/ci/iq/RemoteScanner.groovy [hunk 1]
```diff
@@ -84,6 +84,13 @@ class RemoteScanner
     InternalIqClient iqClient = IqClientFactory.getIqLocalClient(log, instanceId)
     def workDirectory = new File(workspace.getRemote())
     def targets = getScanTargets(workDirectory, scanPatterns)
+    for (String pattern : scanPatterns) {
+      if (pattern.startsWith("container:")) {
```
Comment (rasilvap) [inline src/main/java/org/sonatype/nexus/ci/iq/RemoteScanner.groovy:88]: I think "container:" could be a constant

**Comment by rasilvap** [comment]
@bigspotteddog  The build is failing:

```
[ERROR] Failed to execute goal on project nexus-jenkins-plugin: Could not resolve dependencies for project org.sonatype.nexus.ci:nexus-jenkins-plugin:hpi:3.11-SNAPSHOT: Could not find artifact com.sonatype.nexus:nexus-platform-api:jar:3.41.1-01 in incrementals (https://repo.jenkins-ci.org/incrementals/) -> [Help 1]
[2021-07-13T00:54:45.979Z] org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal on project nexus-jenkins-plugin: Could not resolve dependencies for project org.sonatype.nexus.ci:nexus-jenkins-plugin:hpi:3.11-SNAPSHOT: Could not find artifact com.sonatype.nexus:nexus-platform-api:jar:3.41.1-01 in incrementals (https://repo.jenkins-ci.org/incrementals/)
[2021-07-13T00:54:45.979Z]     at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.getDependencies (LifecycleDependencyResolver.java:269)
[2021-07-13T00:54:45.979Z]     at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.resolveProjectDependencies (LifecycleDependencyResolver.java:147)
[2021-07-13T00:54:45.979Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.ensureDependenciesAreResolved (MojoExecutor.java:248)
[2021-07-13T00:54:45.980Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:202)
[2021-07-13T00:54:45.980Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
[2021-07-13T00:54:45.980Z]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
[2021-07-13T00:54:45.980Z]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
[2021-07-13T00:54:45.980Z]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
```

File: src/main/java/org/sonatype/nexus/ci/iq/RemoteScanner.groovy (hunk 1 above)
Comment (bigspotteddog) [reply src/main/java/org/sonatype/nexus/ci/iq/RemoteScanner.groovy]: Using a constant now.

File: src/main/java/org/sonatype/nexus/ci/iq/RemoteScanner.groovy [hunk 2]
```diff
@@ -29,6 +29,9 @@ class RemoteScanner
   static final List<String> DEFAULT_MODULE_INCLUDES =
       ['**/sonatype-clm/module.xml', '**/nexus-iq/module.xml']
 
+  public static final String CONTAINER = "container:"
+
+
```
Comment (koraytugay) [inline src/main/java/org/sonatype/nexus/ci/iq/RemoteScanner.groovy:34]:
```suggestion
```

Comment (bigspotteddog) [reply src/main/java/org/sonatype/nexus/ci/iq/RemoteScanner.groovy]: That was strange. I used IntelliJ create constant to do have it make that change and it must have added an extra line?

**Review by rasilvap — APPROVED** [review]
The pr looks good to me.

**Review by koraytugay — APPROVED** [review]
+1

**Review by koraytugay — APPROVED** [review]
+1 Approve Again

**Pull request merged**